File: app/src/main/java/ave/mujica/poc/bglog/BackgroundLogResult.java
package ave.mujica.poc.bglog;

import java.util.ArrayList;
import java.util.List;

public final class BackgroundLogResult {
    public final String notificationTitle;
    public final String notificationText;
    public final String notificationBigText;
    public final List<String> logLines;

    public BackgroundLogResult(
            String notificationTitle,
            String notificationText,
            String notificationBigText,
            List<String> logLines
    ) {
        this.notificationTitle = notificationTitle;
        this.notificationText = notificationText;
        this.notificationBigText = notificationBigText;
        this.logLines = logLines != null ? new ArrayList<>(logLines) : new ArrayList<>();
    }
}

File: app/src/main/java/ave/mujica/poc/bglog/BackgroundLogResult.kt
package ave.mujica.poc.bglog

class BackgroundLogResult(
    val notificationTitle: String,
    val notificationText: String,
    val notificationBigText: String,
    logLines: List<String>?
) {
    val logLines: List<String> = ArrayList(logLines ?: emptyList())
}